compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -304,6 +526,271 @@ rustc_dep_node_append!([define_dep_nodes!][ <'tcx>
     [] CompileCodegenUnit(Symbol),
 ]);
 
+impl DepKindTrait for dep_kind::Null {
```

Comment:
Can't these traits be implemented by the macro, by adding them to the 
`rustc_dep_node_append!([define_dep_kinds!][ <'tcx> ]);` invokation, 
similarly to how it's done for `define_dep_nodes`?

Reply:
Those `DepKind`s are not associated to queries. This requires a special workaround for `force_from_dep_node` and `try_load_from_on_disk_cache`.